test/org/sagebionetworks/bridge/services/AuthenticationServiceMockTest.java

```diff
@@ -178,6 +202,29 @@ public void signInWithPhone() throws Exception {
         account.setReauthToken(REAUTH_TOKEN);
         doReturn(account).when(accountDao).authenticate(study, PHONE_PASSWORD_SIGN_IN);
         doReturn(PARTICIPANT).when(participantService).getParticipant(study, account, false);
+        doReturn(CONSENTED_STATUS_MAP).when(consentService).getConsentStatuses(any());
+        
+        UserSession retrieved = service.signIn(study, CONTEXT, PHONE_PASSWORD_SIGN_IN);
+        assertEquals(REAUTH_TOKEN, retrieved.getReauthToken());
+    }
+    
+    @Test(expected = ConsentRequiredException.class)
+    public void unconsentedSignInWithPhone() throws Exception {
+        doReturn(account).when(accountDao).authenticate(study, PHONE_PASSWORD_SIGN_IN);
+        doReturn(PARTICIPANT).when(participantService).getParticipant(study, account, false);
+        doReturn(UNCONSENTED_STATUS_MAP).when(consentService).getConsentStatuses(any());
+        
+        service.signIn(study, CONTEXT, PHONE_PASSWORD_SIGN_IN);
+    }
+    
+    @Test
+    public void adminSignInWithPhone() throws Exception {
```

Comment:
test says sign in with phone but method in this test is actually emailSignIn?

Reply:
You can sign in with a phone + password, which is different than the phone-based sign in. The first is a credential + password sign in, while the second is going to send you a code via SMS that you can enter, in order to avoid having a password.